astropy/modeling/functional_models.py

```diff
@@ -2014,3 +2014,116 @@ def evaluate(cls, x, y, amplitude, r_eff, n, x_0, y_0, ellip, theta):
+        the positive x-axis.
+    x_0 : float, optional
+        x position of the center.
+    y_0 : float, optional
+        y position of the center.
+
+    Notes
+    -----
+    Model formula:
+
+    .. math::
+
+        f(x) = \\sum_n f_n B_n (x; \\beta)
+
+    for :math:`f(x)` the flux of an object and
+    :math:`B_n (x; \\beta) = B_{n_1} (x_1; \\beta) B_{n_2} (x_2; \\beta)`,
+    where
+
+    .. math::
+    
+        B_n (x; \\beta) = \\left[2^n \\pi ^ {1/2} n! \\beta \\right]^{-1/2} H_n \\left( \\frac{x}{\\beta} \\right) e^{-\\frac{x^2}{2\\beta^2}}
+
+    (where :math:`H_n(x)` is a Hermite polynomial of order :math:`n`) and
+
+    .. math::
+
+        f_n = \\int_{-\\infty} ^{\\infty} d^2 x \\ f(x) B_n (x; \\beta)
+
+    Examples
+    --------
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        from astropy.modeling.models import CartesianShapelets
+        import matplotlib.pyplot as plt
+
+        x,y = np.meshgrid(np.arange(31), np.arange(31))
+        mod = CartesianShapelets(beta0=2, beta1=2, nmax=6, phi=0,
+            x_0=15, y_0=15,)
+        img = mod(x, y) # returns gaussian by default
+
+        plt.figure()
+        plt.imshow(img, origin='lower', interpolation='nearest')
+        plt.xlabel('x')
+        plt.ylabel('y')
+        plt.show()
+
+    References
+    ----------
+    .. [1] Refregier, A. 2003, MNRAS 338, 35
+    .. [2] Refregier, A., & Bacon, D. 2003, MNRAS 338, 48
+    """
+
+    beta0 = Parameter(default=2.)
+    beta1 = Parameter(default=2.)
+    nmax = Parameter(default=6)
+    phi = Parameter(default=0)
+    x_0 = Parameter(default=0)
+    y_0 = Parameter(default=0)
+    _hermite = None
+
+    @classmethod
+    def evaluate(cls, x, y, beta0, beta1, nmax, phi, x_0, y_0, ft=False):
+        from scipy.misc import factorial
+        basis_func_arr = []
+        if cls._hermite == None:
+            try:
+                from scipy.special import hermite
+            except ValueError:
+                raise ImportError('CartesianShapelets model requires scipy > 0.11.')
+
+        for n0 in range(nmax):
+            for n1 in range(nmax):
+                h0 = hermite(n0)
+                h1 = hermite(n1)
+                rot_mat = np.array([[np.cos(phi),-1.*np.sin(phi)],[np.sin(phi),np.cos(phi)]]) # rotation matrix
+                const0 = 1
+                const1 = 1
+                if ft: # compute Fourier Transform
+                    beta0 = 1 / beta0
+                    beta1 = 1 / beta1
+                    const0 = 1j ** n0
+                    const1 = 1j ** n1
+                h0 *= const0 * (beta0 ** -.5) * ((2 ** n0) * (np.pi ** 0.5) * factorial(n0)) ** -.5
+                h1 *= const1 * (beta1 ** -.5) * ((2 ** n1) * (np.pi ** 0.5) * factorial(n1)) ** -.5
+                func0 = lambda z: h0(z / beta0) * np.exp(-.5 * ((z / beta0) ** 2))
+                func1 = lambda z: h1(z / beta1) * np.exp(-.5 * ((z / beta1) ** 2))
```

Comment:
The only difference here is `hx` and `betax`. So instead of two lambda functions you could just create one function which takes `h` and `beta` as parameters.